### PR TITLE
Flow and quoted verb values reach the full parser; a menu at the top edge keeps its margin

### PR DESCRIPTION
**Pull request by h3xxit**
Both findings cubic raised on the release PR (#114), fixed on `dev` so the release carries them.

**Access grammar (P2, on #113's fallback).** The subset parser *succeeds* on `read: [A, B]` and `read: "A <a@x>"` — it passes the brackets or quotes to the entry parser as text, which drops the entry, or worse keeps the quotes as part of a role name (`'Developer'` resolved to the role `'developer'`). So the fallback never fired for these. A verb value that looks like syntax the subset parser passed through (flow sequence/mapping, quoted scalar — inline or as a list item) now sends the document to the full parser; a document the full parser rejects keeps the subset answer. Tests: flow sequence, quoted inline and in a block list, and the plain forms staying on the subset path. Access suites: 439 green.

**Menu placement (P3).** The downward branch placed the menu at the pointer's own `y` even inside the top margin. Clamped to `MENU_MARGIN` and re-checked for fit at the clamped position; the other branches are unchanged. New hook test covers the four placements and the top-edge case.

Changeset: patch on core-backend and core-frontend.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
## Summary by cubic
Fixes two release findings: flow-sequence and quoted access verbs now resolve to the grants they spell, and a menu opened at the top edge keeps the same margin as every other placement.

**Bug Fixes**
- Verbs like `read: [A, B]` or `read: "A <a@x>"` previously passed their brackets or quotes to the entry parser as text, dropping the entry or keeping quotes in the role name; now they reach the full parser, and a document it rejects keeps the subset answer.
- The full parser now uses the failsafe schema, so scalars stay the text they were written as — `0x10` stays `0x10`, not `16` — and plain entries beside a quoted one keep resolving exactly as on the subset path.
- The downward menu placement clamps the pointer's y to the margin and re-checks fit there.
- Adds tests for quoted and flow values and the top-edge placement, and ships a patch changeset for `@bevel-software/platform-core-backend` and `@bevel-software/platform-core-frontend`.

<sup>Written for commit 0ed9a56d35dba96ac0443a757eb63d1845382bf8. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/Bevel-Software/Hexis/pull/115?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/core-backend/src/modules/access-model/access-grammar.ts [hunk 1]
```diff
@@ -817,14 +817,38 @@ export type OwnEntries = Record<Verb, ParsedEntry[]>;
...
 function ownEntriesRoot(frontmatter: string): unknown {
   const subset = parseYamlSubset(frontmatter);
-  if (subset.ok) return subset.value;
+  if (subset.ok && !verbValuesNeedFullYaml(subset.value)) return subset.value;
   try {
-    return parseFullYaml(frontmatter);
+    const full = parseFullYaml(frontmatter);
+    // A document the full parser rejects but the subset parser accepted keeps
+    // the subset answer — whatever it read is what has always been read.
+    return full ?? (subset.ok ? subset.value : null);
```
[the remark this answers is not in the document]
Comment (h3xxit) [reply packages/core-backend/src/modules/access-model/access-grammar.ts]: Valid — fixed in 15a69c6: when the full parser is used, verb values are normalised back to text (scalars and list items; nested mappings stay for the entry parser to refuse as before). Test pins that `write: true` / `- 42` resolve identically beside a quoted value and on the subset path.

File: packages/core-backend/src/modules/access-model/access-grammar.ts [hunk 2]
```diff
@@ -817,14 +817,63 @@ export type OwnEntries = Record<Verb, ParsedEntry[]>;
...
+ * handed the entry grammar text. A principal is text, so a verb value is put
+ * back into the form every entry has always arrived in: scalars become their
+ * source text, list items likewise, and a nested mapping (which the grammar
+ * never accepted) is left for the entry parser to refuse as before.
+ */
+function verbValuesAsText(root: unknown): unknown {
+  if (!root || typeof root !== 'object' || Array.isArray(root)) return root;
+  const asText = (v: unknown): unknown =>
+    typeof v === 'string' || typeof v === 'number' || typeof v === 'boolean' || typeof v === 'bigint'
+      ? String(v)
```
[the remark this answers is not in the document]
Comment (h3xxit) [reply packages/core-backend/src/modules/access-model/access-grammar.ts]: Valid — fixed in f7f283d: the fallback now parses with yaml's **failsafe** schema, which never types a scalar, so `0x10`, `1e3`, `true` and dates stay the text they were written as (exactly what the subset parser hands the entry grammar). The stringification pass is gone with it; the test pins `42` / `0x10` / `1e3` resolving to the same role names on both paths.
